tests/e2e/network/service_annotations.go

```diff
@@ -195,6 +196,80 @@ var _ = Describe("Service with annotation", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(ret).To(Equal(true), "external ip %s is not in the target subnet %s", ip, newSubnetCIDR)
 	})
+	It("can be bound to the mode of load balancer", func() {
+		mode_name := "validate_lb_mode"
+		annotation := map[string]string{
+			azure.ServiceAnnotationLoadBalancerMode: mode_name,
+		}
+
+		service := createLoadBalancerServiceManifest(cs, serviceName, annotation, labels, ns.Name, ports)
+		_, err := cs.CoreV1().Services(ns.Name).Create(service)
+		Expect(err).NotTo(HaveOccurred())
+		utils.Logf("Successfully created LoadBalancer service " + serviceName + " in namespace " + ns.Name)
+
+		if mode_name == "" {
+			utils.Logf("Service mode not set or set to default")
+		}else if mode_name == "auto" {
+			utils.Logf("Service mode set to auto")
+		}else{
+		utils.Logf("Service mode configured to %s",mode_name)
+		}
+		//get nodelist
+		v,err := utils.GetAgentNodes(cs)
+		//get providerID specific to an agentnode
+		Expect(err).NotTo(HaveOccurred())
+			if len(v) >= 2 {
+				vmss_name := make([]string, len(v))
+				for j :=0; j <len(v); j++ {
+					node := v[j]
+		            providerID := node.Spec.ProviderID
+		            if providerID != "" {
+					    utils.Logf("ProviderID for the node "+providerID)
+					    i := strings.Index(providerID,"k8s")
+					    if i != -1 {
+					        vmss_name[j] = providerID[i:][:strings.Index(providerID[i:],"/virtualMachines/")]
+					    }
+					}
+				}
+				//selecting two vmss names from the slice which are not same
```

Comment:
After switching to sets based on above comments, check length of `vmssNames`. If its length is <= 2, then skip the tests and exit

Reply:
then you can just refer vmssNames beblow, e.g. validateLoadBalancerBackendPools(vmssName[0]) and validateLoadBalancerBackendPools(vmssName[1])